### ISSUE TITLE
[terra-infinite-list] Accessibility Issues With Virtualization and Scroll-based Loading

### ISSUE THREAD

**Issue by dkasper-was-taken** [cerner/terra-framework#1043]
# Bug Report
Accessibility Issues With Virtualization and Scroll-based loading of ListItems

## Description
InfiniteList is missing the appropriate aria labels for row indexes and row counts to describe to a user which components they are viewing.

InfiniteList doesn't maintain tabbing and arrow based navigation focus using assistive technology if the next available list item or tab index has been virtualized. This results in elements at the end of the list being unreachable via keyboard.

## Expected Behavior
Using AT keys for list item cycling listen for incorrectly defined row indexes.

Using the tab key should be able to cycle through all selectable items and sub items even if they have been virtualized.

## Possible Solution
The concept of paging/virtualization doesn't exist with html list, only tables. So rebuilding with the new terra-table may be the only way.

ListItem can only contain read-only content and focus of elements needs to be brute forced.

## @ Mentions
@neilpfeiffer

**Comment by ryanthemanuel** [comment]
@scottwilmarth we will probably need to discuss this to see what options we have (if any).  If there aren't any, we need to move away from using this component (possibly even deprecating it).

**Comment by scottwilmarth** [comment]
@ryanthemanuel infinite scrolling presents many challenges to people with disabilities, especially those using assistive technologies. Generally speaking, infinite scrolling should be used for discovering new things. It should not be used for scrolling a definitive list the user needs to review or find something in. From an accessibility standpoint, I do not recommend it at all. From a UX perspective, I do not think it has a role in Cerner's enterprise software. I will follow up with UX Foundations for their feedback.

**Comment by ryanthemanuel** [comment]
Per @scottwilmarth's comments I'm closing this issue as won't fix.  We will be deprecating terra-infinite-list.  Expect a doc/npm update shortly with explanations and suggestions.
